File: prosessering/src/test/java/no/nav/familie/prosessering/task/Task1.java
package no.nav.familie.prosessering.task;

import no.nav.familie.prosessering.AsyncTask;
import no.nav.familie.prosessering.TaskBeskrivelse;
import no.nav.familie.prosessering.domene.Task;
import no.nav.familie.prosessering.domene.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.concurrent.TimeUnit;

@Service
@TaskBeskrivelse(taskType = Task1.TASK_1, beskrivelse = "Dette er task 1")
public class Task1 implements AsyncTask {

    public static final String TASK_1= "task1";

    private TaskRepository taskRepository;


    @Autowired
    public Task1(TaskRepository taskRepository) {
        this.taskRepository = taskRepository;
    }


    @Override
    public void doTask(Task task) {

        try {
            TimeUnit.MICROSECONDS.sleep(1);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onCompletion(Task task){
        Task nesteTask = Task.nyTask(Task2.TASK_2, task.getPayload());
        taskRepository.save(nesteTask);
    }
}

File: prosessering/src/test/java/no/nav/familie/prosessering/task/Task1.kt
package no.nav.familie.prosessering.task

import no.nav.familie.prosessering.AsyncTask
import no.nav.familie.prosessering.TaskBeskrivelse
import no.nav.familie.prosessering.domene.Task
import no.nav.familie.prosessering.domene.Task.Companion.nyTask
import no.nav.familie.prosessering.domene.TaskRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.concurrent.TimeUnit

@Service @TaskBeskrivelse(tasktype = Task1.TASK_1, beskrivelse = "Dette er task 1")
class Task1 @Autowired constructor(private val taskRepository: TaskRepository) : AsyncTask {


    override fun doTask(task: Task) {
        try {
            TimeUnit.MICROSECONDS.sleep(1)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }

    override fun preCondition(task: Task) {
        super.preCondition(task)
    }

    override fun onCompletion(task: Task) {
        val nesteTask =
                nyTask(Task2.TASK_2, task.payloadId)
        taskRepository.save(nesteTask)
    }

    companion object {
        const val TASK_1 = "task1"
    }

}
